core-bundle/src/OptIn/OptInTokenInterface.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\OptIn;
+
+interface OptInTokenInterface
+{
+    /**
+     * Returns the token identifier.
+     */
+    public function getIdentifier(): string;
+
+    /**
+     * Returns the e-mail address.
+     */
+    public function getEmail(): string;
+
+    /**
+     * Returns true if the token is valid.
+     */
+    public function isValid(): bool;
+
+    /**
+     * Confirms the token.
+     */
+    public function confirm(): void;
```

Comment:
The exceptions that can be thrown should be documented with `@throws` and ideally they should be special exceptions. E.g. `TokenAlreadyConfirmedException`.